packages/engine/src/lib/helper/trigger-helper.ts

```diff
@@ -139,94 +141,102 @@ export const triggerHelper = {
+                        message: `Error while testing trigger: ${inspect(handshakeResponseError)}`,
                     }
                 }
+                return {
+                    success: true,
+                    response: handshakeResponse,
+                }
             }
-            case TriggerHookType.TEST:
-                try {
-                    return {
-                        success: true,
-                        output: await pieceTrigger.test({
-                            ...context,
-                            files: createFilesService({
-                                apiUrl: constants.internalApiUrl,
-                                engineToken: params.engineToken!,
-                                stepName: triggerName,
-                                flowId: params.flowVersion.flowId,
-                            }),
+            case TriggerHookType.TEST: {
+                const { data: testResponse, error: testResponseError } = await tryCatch((async () => {
+                    return pieceTrigger.test({
+                        ...context,
+                        files: createFilesService({
+                            apiUrl: constants.internalApiUrl,
+                            engineToken: params.engineToken!,
+                            stepName: triggerName,
+                            flowId: params.flowVersion.flowId,
                         }),
-                    }
-                }
-                catch (e) {
+                    })
+                })())
+
+                if (testResponseError) {
+                    console.error(testResponseError)
                     return {
                         success: false,
-                        message: `Error while testing trigger: ${inspect(e)}`,
+                        message: `Error while testing trigger: ${inspect(testResponseError)}`,
                         output: [],
                     }
                 }
+                return {
+                    success: true,
+                    output: testResponse,
+                }
+            }
             case TriggerHookType.RUN: {
                 if (pieceTrigger.type === TriggerStrategy.APP_WEBHOOK) {
-                    if (!params.appWebhookUrl) {
-                        throw new Error(`App webhook url is not available for piece name ${pieceName}`)
-                    }
-                    if (!params.webhookSecret) {
-                        throw new Error(`Webhook secret is not available for piece name ${pieceName}`)
-                    }
 
-                    try {
-                        const verified = piece.events?.verify({
+                    const { data: verified, error: verifiedError } = await tryCatch((async () => {
+                        if (!params.appWebhookUrl) {
+                            throw new Error(`App webhook url is not available for piece name ${pieceName}`)
```

Comment:
I think these must be a EngineGenricError (of type engine) instead of normal error